### PR TITLE
[TE] Fix ignored timeout by checking Thread.interrupted

### PR DESCRIPTION
**Pull request by vincentchenjl**

### CODE REVIEW

File: thirdeye/thirdeye-pinot/src/main/java/org/apache/pinot/thirdeye/detection/algorithm/DimensionWrapper.java [hunk 1]
```diff
@@ -249,6 +249,12 @@ public DetectionPipelineResult run() throws Exception {
       checkEarlyStop(totalNestedMetrics, successNestedMetrics, i, lastException);
       MetricEntity metric = nestedMetrics.get(i);
       try {
+        if (Thread.interrupted()) {
```
Comment (jihaozh) [inline thirdeye/thirdeye-pinot/src/main/java/org/apache/pinot/thirdeye/detection/algorithm/DimensionWrapper.java]: Mind add some comment here specific when will the thread get interrupted? For example, when timing out or shutdown.

Comment (vincentchenjl) [reply thirdeye/thirdeye-pinot/src/main/java/org/apache/pinot/thirdeye/detection/algorithm/DimensionWrapper.java]: Just added the comment.

Comment (xiaohui-sun) [reply thirdeye/thirdeye-pinot/src/main/java/org/apache/pinot/thirdeye/detection/algorithm/DimensionWrapper.java]: Please also add more context like below "Detection with multiple dimensions may take a significantly long time, which would be cancelled from TaskDriver."

File: thirdeye/thirdeye-pinot/src/main/java/org/apache/pinot/thirdeye/detection/algorithm/DimensionWrapper.java [hunk 2]
```diff
@@ -249,6 +249,13 @@ public DetectionPipelineResult run() throws Exception {
...
+          // this happens when Future.cancel() or ExecutorService.shutdown() is called.
+          LOG.error("Current thread is interrupted before running dimension {}/{} for detection {} metrics {}, "
+                  + "so skip the rest of dimensions and return the intermediate",
```
Comment (xiaohui-sun) [inline thirdeye/thirdeye-pinot/src/main/java/org/apache/pinot/thirdeye/detection/algorithm/DimensionWrapper.java]: we don't need  the first "+".

File: thirdeye/thirdeye-pinot/src/main/java/org/apache/pinot/thirdeye/detection/algorithm/DimensionWrapper.java [hunk 3]
```diff
@@ -249,6 +249,13 @@ public DetectionPipelineResult run() throws Exception {
...
+              i + 1 , totalNestedMetrics, this.config.getId(), metric.getUrn());
+          break;
```
Comment (xiaohui-sun) [inline thirdeye/thirdeye-pinot/src/main/java/org/apache/pinot/thirdeye/detection/algorithm/DimensionWrapper.java]: Add a comment: "We still want to keep the current detection result and store it anyway."

**Pull request merged**